crates/client/rpc/src/lib.rs

```diff
@@ -137,7 +137,10 @@ where
     pub fn current_block_hash(&self) -> Result<H256, ApiError> {
         let substrate_block_hash = self.client.info().best_hash;
 
-        let starknet_block = get_block_by_block_hash(self.client.as_ref(), substrate_block_hash).unwrap_or_default();
+        let starknet_block =
+            get_block_by_block_hash(self.client.as_ref(), substrate_block_hash).ok_or(ApiError::UnknownBlock(
```

Comment:
And rather use StarknetAPiError

Reply:
@tdelabro Thank you for your comment. Do you mean if get_block_by_block_hash returns Err here, just ignore this error and don't propagate this error?

> ```
> pub fn get_block_by_block_hash<B, C>(client: &C, block_hash: <B as BlockT>::Hash) -> Option<StarknetBlock>
> where
>     B: BlockT,
>     C: HeaderBackend<B>,
> {
>     let header = client.header(block_hash).ok().flatten()?;
>     let digest = header.digest();
>     let block = find_starknet_block(digest).ok()?;
>     Some(block)
> }
> ```
> 
> Change the signature of `get_block_by_block_hash` so that it returns a Result. We are ignoring the errors here.

@tdelabro Thank you for your comment. Do you mean if `get_block_by_block_hash` returns `Err` here, just ignore this error and don't propagate this error? Thanks

Reply:
No I meant that the current behaviour is that we are ignoring the error. Changing them to `None` with `ok()`. We should bubble those errors up instead